inspector/src/components/graph/canvasGraphService.ts

```diff
@@ -38,17 +39,222 @@ export class CanvasGraphService {
+
+        this._drawTimeAxis(globalTimeMinMax, drawableArea);
+
+        datasets.forEach((dataset: IPerfDataset) => {
+            const valueMinMax = this._getMinMax(dataset.data.map((point: IPerfPoint) => point.value));
+            const drawablePoints = dataset.data.map((point: IPerfPoint) => this._getPixelPointFromDataPoint(point, globalTimeMinMax, valueMinMax, drawableArea));
+
+            let prevPoint: IPerfPoint = drawablePoints[0];
+            ctx.beginPath();
+            
+            drawablePoints.forEach((point: IPerfPoint) => {
+                ctx.moveTo(prevPoint.timestamp, prevPoint.value);
+                ctx.lineTo(point.timestamp, point.value);
+                prevPoint = point;
+            });
+            ctx.stroke();
         });
+    }
+
+    /**
+     * Draws the time axis, adjusts the drawable area for the graph.
+     * 
+     * @param timeMinMax the minimum and maximum for the time axis. 
+     * @param drawableArea the current allocated drawable area. 
+     */
+    private _drawTimeAxis(timeMinMax: IPerfMinMax, drawableArea: IGraphDrawableArea) {
+        const { _ctx: ctx } = this;
+
+        if (!ctx) {
+            return;
+        }
+        const spaceAvailable = drawableArea.right - drawableArea.left;
 
-        // TODO: Remove this code after dataset drawing is implemented.
+        const ticks = this._getTicks(timeMinMax, spaceAvailable);
+
+        const axisHeight = 100;
+
+        // remove the height of the axis from the available drawable area.
+        drawableArea.bottom -= axisHeight;
+
+        // draw time axis line
+        ctx.save();
         ctx.beginPath();
-        ctx.moveTo(100, 100);
-        ctx.lineTo(200, 0);
+        ctx.moveTo(drawableArea.left, drawableArea.bottom);
+        ctx.lineTo(drawableArea.right, drawableArea.bottom);
+        
+        // draw ticks and text.
+        ctx.textAlign = "center";
+        ctx.textBaseline = "middle";
+
+        ticks.forEach((tick: number) => {
+            let position = this._getPixelForNumber(tick, timeMinMax, drawableArea.left, spaceAvailable, false);
+            if (position > spaceAvailable) {
+                position = spaceAvailable;
+            }
+            ctx.moveTo(position, drawableArea.bottom);
+            ctx.lineTo(position, drawableArea.bottom + 10);
+            ctx.fillText(tick.toString(), position, drawableArea.bottom + 20);
+        });
         ctx.stroke();
-    }    
+        ctx.restore();
+    }
+
+
+    /**
+     * Gets a list of ticks given the min and max of the axis, and the space available in the axis.
+     * 
+     * @param minMax the minimum and maximum values of the axis
+     * @param spaceAvailable the total amount of space we have allocated to our axis
+     * @returns a list of "nice" tickS.
+     */
+    private _getTicks(minMax: IPerfMinMax, spaceAvailable: number): number[] {
+        const { min, max } = minMax;
+        const minTickSpacing = 40; 
+
+        let ticks: number[] = [];

```

Comment:
maybe try reusing this array instead of creating it every time you draw the graph? This will help with garbage collection and will support realtime rendering

Reply:
I think I could reduce the times I need to recreate the array, but probably cant get away with reusing the array every time. 

I could potentially reuse the array when oldNiceMin === currNiceMin && oldNiceMax === currNiceMax
(Perhaps also && oldSpacing === spacing because the loop depends on spacing as well).

When any of those are different I'll probably need to create a fresh array.

Do you think this is a good solution to the problem?

Reply:
can't you reuse the array while emptying it on each frame? setting the array's length to 0 is a quick operation which clears the array. so instead of `let ticks: number[] = [];` you just do `this.ticksArray.length = 0`

Reply:
Oooo okay I was under the impression that the performance implications for both those operations were the same!

I will change this to .length = 0 instead :D

Reply:
Done!